// @atlaskit/logo/dist/cjs/opsgenie-logo/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "OpsGenieIcon", {
  enumerable: true,
  get: function get() {
    return _oldIcon.OpsGenieIcon;
  }
});
Object.defineProperty(exports, "OpsGenieLogo", {
  enumerable: true,
  get: function get() {
    return _oldLogo.OpsGenieLogo;
  }
});
Object.defineProperty(exports, "OpsGenieWordmark", {
  enumerable: true,
  get: function get() {
    return _oldWordmark.OpsGenieWordmark;
  }
});
Object.defineProperty(exports, "OpsgenieIcon", {
  enumerable: true,
  get: function get() {
    return _newIcon.OpsgenieIcon;
  }
});
Object.defineProperty(exports, "OpsgenieLogo", {
  enumerable: true,
  get: function get() {
    return _newLogo.OpsgenieLogo;
  }
});
Object.defineProperty(exports, "OpsgenieWordmark", {
  enumerable: true,
  get: function get() {
    return _newWordmark.OpsgenieWordmark;
  }
});
var _oldIcon = require("./old-icon");
var _newIcon = require("./new-icon");
var _oldLogo = require("./old-logo");
var _newLogo = require("./new-logo");
var _oldWordmark = require("./old-wordmark");
var _newWordmark = require("./new-wordmark");
